core/orch_test.go

```diff
@@ -708,27 +700,88 @@ func TestProcessPayment_GivenReceiveTicketError_ReturnsError(t *testing.T) {
 	recipient.AssertNumberOfCalls(t, "RedeemWinningTicket", 2)
 }
 
-func TestProcessPayment_GivenInsufficientCredit_ReturnsErr(t *testing.T) {
+// Check that an Acceptable error increases the credit
+
+func TestProcessPayment_AcceptablePaymentError_IncreasesCreditBalance(t *testing.T) {
 	n, _ := NewLivepeerNode(nil, "", nil)
 	n.Balances = NewBalances(5 * time.Second)
 	recipient := new(pm.MockRecipient)
 	n.Recipient = recipient
 	orch := NewOrchestrator(n)
 	manifestID := ManifestID("some manifest")
-	sessionID := "some sessionID"
-	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return(sessionID, true, nil)
-	recipient.On("EV").Return(big.NewRat(1000000000, 1))
+	acceptableError := pm.NewMockReceiveError(errors.New("Acceptable ReceiveTicket error"), true)
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", false, acceptableError).Once()
+	assert := assert.New(t)
+	// Create a ticket where faceValue = EV instead of using 'defaultTicket'
+	// FaceValue is the expected balance increase
+	faceVal := big.NewInt(100)
```

Comment:
Why do we need to do all this boilerplate setup for generating a payment? Couldn't we just use `defaultPayment()` and then calculate the EV based on the ticket params included in the returned payment? These questions apply to the other tests in this commit that do a similar setup for generating a payment

Reply:
You're saying calling the mock for "EV" with the faceValue/winProb that is generated by defaultPayment?

Could do that, yes.